src/Datadog.Trace.ClrProfiler.Native/cor_profiler.cpp

```diff
@@ -17,58 +18,69 @@ namespace trace {
 
 CorProfiler* profiler = nullptr;
 
-CorProfiler::CorProfiler() : integrations_(LoadIntegrationsFromEnvironment()) {
-  Info("CorProfiler::CorProfiler");
-}
+CorProfiler::CorProfiler() { Info("CorProfiler::CorProfiler"); }
 
 HRESULT STDMETHODCALLTYPE
 CorProfiler::Initialize(IUnknown* cor_profiler_info_unknown) {
   is_attached_ = FALSE;
   Info("CorProfiler::Initialize");
+  Info("Environment variables:");
+
+  WSTRING env_vars[] = {environment::tracing_enabled,
+                        environment::debug_enabled,
+                        environment::integrations_path,
+                        environment::process_names,
+                        environment::agent_host,
+                        environment::agent_port,
+                        environment::env,
+                        environment::service_name,
+                        environment::disabled_integrations};
+
+  for (auto&& env_var : env_vars) {
+    Info("  ", env_var, "=", GetEnvironmentValue(env_var));
+  }
 
-  Info(kIntegrationsEnvironmentName, " ",
-       GetEnvironmentValue(kIntegrationsEnvironmentName));
-  Info("DD_AGENT_HOST"_W, GetEnvironmentValue("DD_AGENT_HOST"_W));
-  Info("DD_TRACE_AGENT_PORT"_W, GetEnvironmentValue("DD_TRACE_AGENT_PORT"_W));
-  Info("DD_ENV"_W, GetEnvironmentValue("DD_ENV"_W));
-  Info("DD_SERVICE_NAME"_W, GetEnvironmentValue("DD_SERVICE_NAME"_W));
-
-  const auto process_name = GetCurrentProcessName();
+  const WSTRING tracing_enabled =
+      GetEnvironmentValue(environment::tracing_enabled);
 
-  if (integrations_.empty()) {
-    Warn("Profiler disabled: ", kIntegrationsEnvironmentName,
-         " environment variable not set.");
+  if (tracing_enabled == "0"_W || tracing_enabled == "false"_W) {
+    Info("Profiler disabled in ", environment::tracing_enabled);
     return E_FAIL;
   }
 
   const auto allowed_process_names =
-      GetEnvironmentValues(kProcessesEnvironmentName);
-
-  if (allowed_process_names.empty()) {
-    Info(kProcessesEnvironmentName,
-         " environment variable not set. Attaching to any .NET process.");
-  } else {
-    Info(kProcessesEnvironmentName);
-    for (auto& name : allowed_process_names) {
-      Info("  ", name);
-    }
+      GetEnvironmentValues(environment::process_names);
+
+  if (!allowed_process_names.empty()) {
+    const auto process_name = GetCurrentProcessName();
 
     if (std::find(allowed_process_names.begin(), allowed_process_names.end(),
                   process_name) == allowed_process_names.end()) {
-      Info("Profiler disabled: ", process_name, " not found in ", kProcessesEnvironmentName, ".");
+      Info("Profiler disabled: ", process_name, " not found in ",
+           environment::process_names, ".");
       return E_FAIL;
     }
   }
 
+  integrations_ = LoadIntegrationsFromEnvironment();
```

Comment:
Wouldn't it make sense to do `FilterIntegrationsByName` here too? My understanding is that the result will be the same for all modules.

Reply:
I'm not sure why all of these changes are showing up on this PR since they are already in `develop`. But yes, there is a bug here with `FilterIntegrationsByName()` which I am fixing in another PR.